pkg/webhook/topology/add.go

```diff
@@ -0,0 +1,65 @@
+//  Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package topology
+
+import (
+	extensionswebhook "github.com/gardener/gardener/extensions/pkg/webhook"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	"github.com/gardener/gardener-extension-provider-azure/pkg/azure"
+)
+
+const (
+	// WebhookName the name of the topology webhook.
+	WebhookName = "topology"
+	webhookPath = "topology"
+)
+
+var logger = log.Log.WithName("topology-webhook")
+
+// AddToManager creates a webhook adds the webhook to the manager.
+func AddToManager(mgr manager.Manager) (*extensionswebhook.Webhook, error) {
+	logger.Info("Adding webhook to manager")
+
+	types := []extensionswebhook.Type{
+		{Obj: &v1.Pod{}},
```

Comment:
This will register a webhook for pods that reacts for both `CREATE` and `UPDATE` operations (ref: https://github.com/gardener/gardener/blob/219d828fcdea81fb3edf13de2736daf81e137923/extensions/pkg/webhook/registration.go#L290-L291). However, we only want to react on `CREATE` (these fields are anyways immutable on `UPDATE`). With this webhook in place, wouldn't we always run in errors when we try to update existing `Pod`s like this and the request gets rejected by the API server? Would it make sense to allow specifying the operations in the extension library for the webhook registration so that we only register the webhook for the `CREATE` rule?

Reply:
You can technically work around it I believe by not performing changes on requests where `old != nil`. That's not to say that the library shouldn't be enhanced just that it can be done async - not have it a prerequisite for this PR.

Reply:
OK, but I don't see such check in the code. Can you also please open an issue in g/g and add a TODO statement here to remove this special handling once the (to-be-opened) g/g issue is fixed?

Reply:
> OK, but I don't see such check in the code.

Because I just became aware that it is needed :)  

> Can you also please open an issue in g/g and add a TODO statement here to remove this special handling once the (to-be-opened) g/g issue is fixed?

I added a TODO and a test. I will also open the ticket on g/g and assign myself but I would prefer if  for now we are not dependent on next gardener release for consumption. I think that anyway we would need to scan through our other webhooks after and remove this pattern for updates where it is used.